configuration_go/k8sutil/genericdeployment.go

```diff
@@ -35,6 +36,9 @@ type DeploymentGenericConfig struct {
 	ConfigMaps map[string]map[string]string // maps a configmap name to its data of type map[string]string
 	Secrets    map[string]map[string][]byte // maps a secret name to its data of type map[string][]byte
 	Sidecars   []ContainerProvider
+
+	// Post processing of the generated objects (e.g. for adding annotations, modifying monitor namespace, etc.)
+	PostProcess []ObjectProcessor
```

Comment:
I think this is a tricky thing to maintain. The users of this lib will have the freedom to modify the `runtime.Object` map in any way they wish to. 

I think it's just easier for user to get the map, and then modify whatever components they need to in some loop, on _their_ configuration codebase, instead of having to construct functions to pass into struct where they have to conditionally check which element of map to apply to.

Reply:
I agree that you don't need this for basic modifications like changing the serviceMonitor namespace.
But it is trickier when you need to know values defined at the container level.
This is the case that made me do this addition:

- the store container has a side car
- this side car (oauth proxy) consumes a secret name `mytlscerts` that is created by the platform through an annotation on the service
- If I encapsulate the store statefulset configuration in a given function like `makeStore()` and I generate the manifests from another place, I have to return this secret name `mytlscerts` in the function where the `Manifests()` are generated. Because at that point I will be able to do the needed post processing.

How do you think we should handle such cases?

Reply:
Removed. Can be handled nicely by consumers.

Reply:
Yup!